pkgs/games/osu-lazer/update.sh

```diff
@@ -15,7 +16,22 @@ fi
 cd ../../..
 
 if [[ "$1" != "--deps-only" ]]; then
-    update-source-version osu-lazer "$new_version"
+  update-source-version osu-lazer "$new_version"
+  sed -Ei.bak '/ *version = "/s/".+"/"'"$new_version"'"/' "$bin_file"
+  rm "$bin_file.bak"
+  for pair in \
+    'aarch64-darwin osu.app.Apple.Silicon.zip' \
+    'x86_64-darwin osu.app.Intel.zip' \
+    'x86_64-linux osu.AppImage' \
+  ; do
+    set -- $pair
+    nix-prefetch-url "https://github.com/ppy/osu/releases/download/$new_version/$2" | while read -r hash; do
+      nix --extra-experimental-features nix-command hash to-sri --type sha256 "$hash" | while read -r hash; do
+        sed -Ei.bak '/ *'"$1"' = "sha256-/s@".+"@"'"$hash"'"@' "$bin_file"
+        rm "$bin_file.bak"
+      done
+    done
+  done
```

Comment:
Can we have a separate file for the `bin` version? Seems kinda messy to keep two different unrelated code in the same file.